src/main/java/com/wavefront/SpanSender.java

```diff
@@ -2,25 +2,58 @@
 
 import com.wavefront.sdk.common.WavefrontSender;
 
+import java.io.File;
+import java.io.FileWriter;
 import java.io.IOException;
+import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 import java.util.logging.Logger;
 
+import com.google.common.base.Strings;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+
 /**
  * TODO
  *
  * @author Davit Baghdasaryan (dbagdasarya@vmware.com)
  */
 public class SpanSender {
   private static final Logger logger = Logger.getLogger(SpanSender.class.getCanonicalName());
+  private static LoadingCache<String, FileWriter> fileWriterLoadingCache = CacheBuilder.newBuilder()
```

Comment:
I don't see the benefit of using LoadingCache here, spanQueue is written to the file once, so there is no need to keep FileWriter in the cache.

Reply:
Agree